doozer/doozerlib/cli/release_gen_payload.py

```diff
@@ -380,24 +400,30 @@ async def generate_assembly_issues_report(self, assembly_inspector: AssemblyInsp
         # If the assembly claims to have reference nightlies, assert that our payload matches them exactly.
         self.assembly_issues.extend(await PayloadGenerator.check_nightlies_consistency(assembly_inspector))
 
+        span.add_event("Summarizing assembly issue permits")
         return self.summarize_issue_permits(assembly_inspector)
 
+    @TRACER.start_as_current_span("GenPayloadCli.detect_mismatched_siblings")
     def detect_mismatched_siblings(self, assembly_inspector: AssemblyInspector):
         """
         Mismatched siblings are built from the same repo but at a different commit
         """
-
+        span = trace.get_current_span()
         self.logger.debug("Checking for mismatched sibling sources...")
         group_images: List = assembly_inspector.get_group_release_images().values()
+        issues = []
         for mismatched, sibling in PayloadGenerator.find_mismatched_siblings(group_images):
-            self.assembly_issues.append(AssemblyIssue(
```

Comment:
Is this change relevant for the sake of OpenTelemetry?

Reply:
This is an attempt to record assembly issues in span attributes. It is not actually consumed by anything, but I would like to keep this and see how easy or hard to filter those span on visualization systems (e.g. kibana) or monitoring systems.